### PR TITLE
feat: update tokens on login

### PR DESCRIPTION
**Pull request by david972**
Currently Kratos record the token during the registration and the setting.

If the access_token recorded has expired in the meanwhile or tokens has been revoked it would be useful to update them.
I know in case of expiration we can use the refresh token but not in case of revocation.

## Related issue(s)

#1912 

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [x] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [x] I am following the
      [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security. vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [x] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added or changed [the documentation](docs/docs).

### CODE REVIEW

File: selfservice/strategy/oidc/strategy_login.go [hunk 1]
```diff
@@ -109,10 +109,43 @@ func (s *Strategy) processLogin(w http.ResponseWriter, r *http.Request, a *login
 		return nil, s.handleError(w, r, a, provider.Config().ID, nil, errors.WithStack(herodot.ErrInternalServerError.WithReason("The password credentials could not be decoded properly").WithDebug(err.Error())))
 	}
 
+
+	// UPDATE TOKEN
+	var it string
+	if idToken, ok := token.Extra("id_token").(string); ok {
```
Comment (aeneasr) [inline selfservice/strategy/oidc/strategy_login.go:115]: Could you please add a test for this? :)

**Review by aeneasr — CHANGES_REQUESTED** [review]

File: selfservice/strategy/oidc/strategy_login.go (hunk 1 above)
Comment (david972) [reply selfservice/strategy/oidc/strategy_login.go:115]: yes sure

**Comment by david972** [comment]
@aeneasr 
is it miss something ?

**Review by aeneasr — DISMISSED** [review]

**Comment by david972** [comment]
thanks

**Comment by aeneasr** [comment]
@david972 some tests were failing. I merged master and will rerun them, but if they fail, could you please take a look? Otherwise they look good!

It's possible that saving the identity on login in the database has some unwanted side effects, not sure...

**Comment by david972** [comment]
> @david972 some tests were failing. I merged master and will rerun them, but if they fail, could you please take a look? Otherwise they look good!
> 
> It's possible that saving the identity on login in the database has some unwanted side effects, not sure...

sure

**Comment by aeneasr** [comment]
Yeah looks like login tests are failing: https://app.circleci.com/jobs/github/ory/kratos/30259?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

If fixed we can merge :)

**Comment by david972** [comment]
> Yeah looks like login tests are failing: https://app.circleci.com/jobs/github/ory/kratos/30259?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link
> 
> If fixed we can merge :)

ok

**Comment by david972** [comment]
I'll do it first

**Comment by david972** [comment]
Hi
I do not succeed to reproduce the test issue

**Comment by david972** [comment]
I keep looking

**Comment by aeneasr** [comment]
@david972 I am able to reproduce the CI failure with:

```
% cd selfservice/flow/login
% go test -tags sqlite,json1 -short ./...
```

**Comment by david972** [comment]
> @david972 I am able to reproduce the CI failure with:
> 
> ```
> % cd selfservice/flow/login
> % go test -tags sqlite,json1 -short ./...
> ```

Yes I find it

**Comment by david972** [comment]
by the way this call
`a, b := run(t, flow.TypeAPI)`
return error

**Comment by david972** [comment]
find where
I get `unknown hash algorithm` error

**Comment by david972** [comment]
in api flow it did not get the credential

**Comment by david972** [comment]
it's really weird between the two calls the config part disappears

**Comment by AlexandreBrg** [comment]
Hi, my company is also in need of this feature, is it planned to be finished ?

**Comment by david972** [comment]
> Hi, my company is also in need of this feature, is it planned to be finished ?

Hi, need a complete review as this implementation break to many things.
By the way I planned to finish it on 25th April (I guess)

**Comment by david972** [comment]
@aeneasr 
Hi,
If I do a rebase from master can we expect this PR to be merged ?

**Pull request closed without merging**